google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -0,0 +1,264 @@
+      arg -> {
+        BucketOptions.Builder builder = BucketOptions.newBuilder();
+        Explicit.Builder explicitBuilder = Explicit.newBuilder();
+        explicitBuilder.addBounds(0.0D);
+        explicitBuilder.addAllBounds(arg.getBucketBoundaries());
+        builder.setExplicitBuckets(explicitBuilder.build());
+        return builder.build();
+      };
+
+  // promote the following metric labels to monitored resource labels
+  private static final Set<String> PROMOTED_RESOURCE_LABELS =
+      ImmutableSet.of(
+          BuiltinMeasureConstants.PROJECT_ID.getName(),
+          BuiltinMeasureConstants.INSTANCE_ID.getName(),
+          BuiltinMeasureConstants.CLUSTER.getName(),
+          BuiltinMeasureConstants.ZONE.getName(),
+          BuiltinMeasureConstants.TABLE.getName());
+
+  private static final LabelKey CLIENT_UID_LABEL_KEY =
+      LabelKey.create(BuiltinMeasureConstants.CLIENT_UID.getName(), "client uid");
+
+  static com.google.monitoring.v3.TimeSeries convertTimeSeries(
+      MetricDescriptor metricDescriptor,
+      TimeSeries timeSeries,
+      String clientId,
+      MonitoredResource monitoredResource) {
+    String metricName = metricDescriptor.getName();
+    List<LabelKey> labelKeys = metricDescriptor.getLabelKeys();
+    Type metricType = metricDescriptor.getType();
+
+    MonitoredResource.Builder monitoredResourceBuilder = monitoredResource.toBuilder();
+
+    List<LabelKey> metricTagKeys = new ArrayList<>();
+    List<LabelValue> metricTagValues = new ArrayList<>();
+
+    List<LabelValue> labelValues = timeSeries.getLabelValues();
+    for (int i = 0; i < labelValues.size(); i++) {
+      // If the label is defined in the monitored resource, convert it to
+      // a monitored resource label. Otherwise, keep it as a metric label.
+      if (PROMOTED_RESOURCE_LABELS.contains(labelKeys.get(i).getKey())) {
+        monitoredResourceBuilder.putLabels(
+            labelKeys.get(i).getKey(), labelValues.get(i).getValue());
+      } else {
+        metricTagKeys.add(labelKeys.get(i));
+        metricTagValues.add(labelValues.get(i));
+      }
+    }
+    metricTagKeys.add(CLIENT_UID_LABEL_KEY);
+    metricTagValues.add(LabelValue.create(clientId));
+
+    com.google.monitoring.v3.TimeSeries.Builder builder =
+        com.google.monitoring.v3.TimeSeries.newBuilder();
+    builder.setResource(monitoredResourceBuilder.build());
+    builder.setMetric(createMetric(metricName, metricTagKeys, metricTagValues));
+    builder.setMetricKind(createMetricKind(metricType));
+    builder.setValueType(createValueType(metricType));
+    Timestamp startTimeStamp = timeSeries.getStartTimestamp();
+    for (io.opencensus.metrics.export.Point point : timeSeries.getPoints()) {
+      builder.addPoints(createPoint(point, startTimeStamp));
+    }
+    return builder.build();
+  }
+
+  static String getDefaultTaskValue() {
+    String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+    if (jvmName.indexOf(64) < 1) {
```

Comment:
whats 64?

Reply:
Also this would be easier to read with an early return

if (magic) {
  return "java-" + jvmName;
}

String hostname = "localhost";
try {
...}
return       return "java-" + (new SecureRandom()).nextInt() + "@" + hostname;